gr-blocks/lib/moving_average_impl.cc

```diff
@@ -79,6 +81,67 @@ void moving_average_impl<T>::set_scale(T scale)
     d_updated = true;
 }
 
+template <typename T>
+inline void volk_add(T* out, const T* add, unsigned int num)
+{
+    for (unsigned int elem = 0; elem < num; elem++) {
+        out[elem] += add[elem];
+    }
+}
+
+template <>
+inline void volk_add<float>(float* out, const float* add, unsigned int num)
+{
+    volk_32f_x2_add_32f(out, out, add, num);
+}
+
+template <>
+inline void volk_add<gr_complex>(gr_complex* out, const gr_complex* add, unsigned int num)
+{
+    volk_32fc_x2_add_32fc(out, out, add, num);
+}
+
+template <typename T>
+inline void volk_sub(T* out, const T* sub, unsigned int num)
+{
+    for (unsigned int elem = 0; elem < num; elem++) {
+        out[elem] -= sub[elem];
+    }
+}
+
+template <>
+inline void volk_sub<float>(float* out, const float* sub, unsigned int num)
+{
+    volk_32f_x2_subtract_32f(out, out, sub, num);
+}
+
+// TODO: Add volk_sub specialization for gr_complex?
```

Comment:
That would be useful to add this todo. I suggest:
```cpp
template <>
inline void volk_sub<gr_complex>(gr_complex* out, const gr_complex* sub, unsigned int num)
{
    volk_32f_x2_subtract_32f(out, out, sub, 2 * num);
}
```

Reply:
Doh. Of course. Uploading a new version of the commit as soon as I've run the tests.

Reply:
Added som paranoia compile time tests.